### ISSUE TITLE
Postgres `SELECT ... LIMIT 1 FOR UPDATE` unparsable

### ISSUE THREAD

**Issue by kgutwin** [sqlfluff/sqlfluff#6425]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The Postgres docs (https://www.postgresql.org/docs/current/sql-select.html) seem to suggest that the row-level locking syntax `FOR UPDATE` follows after `LIMIT` clauses. However, SQLFluff fails to lint the query `SELECT foo FROM bar LIMIT 1 FOR UPDATE`. This query syntax is acceptable to Postgres, as is the opposite order `SELECT foo FROM bar FOR UPDATE LIMIT 1`. SQLFluff only accepts this latter structure, though.

### Expected Behaviour

Linting successful with no parsing errors.

### Observed Behaviour

```
== [stdin] FAIL                                                                                                                                                
L:   1 | P:  29 |  PRS | Line 1, Position 29: Found unparsable section: 'FOR
                       | UPDATE'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```bash
echo 'SELECT foo FROM bar LIMIT 1 FOR UPDATE' | sqlfluff lint --dialect=postgres -
```

### Dialect

postgres

### Version

sqlfluff, version 3.2.5
Python 3.11.10

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
